source/configuration/modules/imfile.rst

```diff
@@ -1,570 +1,408 @@
 
-   This option can be used when ``readMode`` is 0 or 2.
+* /path/to/file.log is valid and will monitor file named file.log
+* /path/to/file\*.log is valid and will monitor all files named like *file\**.log
+* /path/to/\*name.log is valid and will monitor all files named like *\*name*.log
+* /path/to\*/file.log is invalid
+* /path/to/foo being foo a directory is also invalid
 
-.. index::
-   single: imfile; freshStartTail
-.. function:: freshStartTail [on/off] (requires v8.18.0+)
+============  =============
+**REQUIRED**
+tag:          "value"
+Default:      none
+Legacy:       $InputFileTag
+============  =============
 
-   **Default: off**
+**TODO: preperly link Syslog Specification https://tools.ietf.org/html/rfc5424#section-6.2.5**
 
-   This is used to tell rsyslog to seek to the end/tail of input files
-   (discard old logs)**at its first start(freshStart)** and process only new 
-   log messages.
-   
-   When deploy rsyslog to a large number of servers, we may only care about 
-   new log messages generated after the deployment. set **freshstartTail**
-   to **on** will discard old logs. Otherwise, there may be vast useless
-   message burst on the remote central log receiver
+The tag or application name that originated the file (message), according to RFC. If you would like to see the colon after the tag, you need to specify it here (like 'tag="myTagValue:"').
+ 
+=========  ==================
+facility:  "value"
+Default:   "local0"
+Legacy:    $InputFileFacility
+=========  ==================
 
-Caveats/Known Bugs
-------------------
+**TODO: preperly link Syslog Specification https://tools.ietf.org/html/rfc5424#section-6.2.5**
 
-* currently, wildcards are only supported in inotify mode
-* read modes other than "0" currently seem to have issues in
-  inotify mode
+The syslog facility to be assigned to lines read. Textual form is suggested (e.g. "local0", "local1", ...) but numbers are also accepted (e.g. 128 for "local0") as defined in RFC.
 
-Configuration Example
----------------------
+=========  ==================
+severity:  "value"
+Default:   "notice"
+Legacy:    $InputFileSeverity
+=========  ==================
 
-The following sample monitors two files. If you need just one, remove
-the second one. If you need more, add them according to the sample ;).
-This code must be placed in /etc/rsyslog.conf (or wherever your distro
-puts rsyslog's config files). Note that only commands actually needed
-need to be specified. The second file uses less commands and uses
-defaults instead.
+**TODO: preperly link Syslog Specification https://tools.ietf.org/html/rfc5424#section-6.2.5**
 
-::
+The syslog severity to be assigned to lines read. Textual form is suggested (e.g. "info", "warning", ...) but numbers are also accepted (e.g. 4 for "info") as defined in RFC.
 
-  module(load="imfile" PollingInterval="10") #needs to be done just once 
+========  =====================
```

Comment:
add a link here to the ruleset documentation.

This is borderline expert mode.